### PR TITLE
TSK-1227: made ObjectReference#System, Objectreference#SystemInstance…

### PR DESCRIPTION
**Pull request by sfrevel**
… optional, allow null in database

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java [hunk 1]
```diff
@@ -632,32 +650,6 @@ void testThrowsExceptionIfAttachmentIsInvalid() throws ClassificationNotFoundExc
...
-            createObjectReference(
-                "COMPANY_A",
-                null,
-                "INSTANCE_B",
-                "ArchiveId",
-                "12345678901234567890123456789012345678901234567890"),
-            "E-MAIL",
-            "2018-01-15",
-            createSimpleCustomProperties(3)));
-
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java:660]: Why did you remove this?

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java [hunk 2]
```diff
@@ -766,16 +758,18 @@ void testThrowsExceptionIfMandatoryPrimaryObjectReferenceIsNotSetOrIncomplete()
         createObjectReference("COMPANY_A", "SYSTEM_A", "INSTANCE_A", "VNR", null));
     testCreateTask.accept(
         createObjectReference("COMPANY_A", "SYSTEM_A", "INSTANCE_A", null, "1234567"));
-    testCreateTask.accept(createObjectReference("COMPANY_A", "SYSTEM_A", null, "VNR", "1234567"));
-    testCreateTask.accept(createObjectReference("COMPANY_A", null, "INSTANCE_A", "VNR", "1234567"));
+    //    testCreateTask.accept(createObjectReference("COMPANY_A", "SYSTEM_A", null, "VNR",
+    // "1234567"));
+    //    testCreateTask.accept(createObjectReference("COMPANY_A", null, "INSTANCE_A", "VNR",
+    // "1234567"));
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java:764]: Why commented out?

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java (hunk 1 above)
Comment (sfrevel) [reply lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java:660]: Because the test expected an exception which is not thrown any more when ObjecReference#System/System_Instance is not set.

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java (hunk 2 above)
Comment (sfrevel) [reply lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java:764]: Same as above. but should be deleted not commented.

Because the test expected an exception which is not thrown any more when ObjecReference#System/System_Instance is not set.

**Comment by sfrevel** [comment]
> Did you update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)?

Where can I find the source of the overview picture https://taskana.atlassian.net/wiki/spaces/TAS/pages/200572994/Data+Model#DataModel-Overview to edit the column which became nullable?

> Did you update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)?

Where can I find the source of the Diagram for removing the NOT NULLs https://taskana.atlassian.net/wiki/spaces/TAS/pages/200572994/Data+Model#DataModel-Overview

**Comment by sfrevel** [comment]
> Hey @sfrevel!
> Can I trust you that the migration scripts are working? Did you try it out with every database?

Thanks for reminding me!
